src/java.base/share/classes/java/lang/reflect/code/op/ExtendedOps.java

```diff
@@ -2791,6 +2792,107 @@ public Value processor() {
+                Block.Builder bb = builders.get(exprIndex);
+                if (exprIndex == 0) {
+                    b.op(branch(bb.successor()));
+                }
+                bb.transformBody(expression, List.of(), opT.andThen((block, op) -> {
+                    if (op instanceof YieldOp yop) {
+                        expressions.add(block.context().getValue(yop.yieldValue()));
+                        block.op(branch(builders.get(exprIndex + 1).successor()));
+                    } else if (op instanceof Lowerable lop) {
+                        block = lop.lower(block, opT);
+                    } else {
+                        block.op(op);
+                    }
+                    return block;
+                }));
+            }
+
+            Block.Builder builder = builders.getLast();
+
+            MethodRef listOfArray = MethodRef.method(JavaType.J_U_LIST, "of", JavaType.J_U_LIST, J_L_OBJECT_ARRAY);
+
+            Op.Result fragmentsList = builder.op(invoke(listOfArray,
+                    buildArrayObject(builder, builder.context().getValues(fragments()))));
+
+            Op.Result expressionsList = builder.op(invoke(listOfArray, buildArrayObject(builder, expressions)));
+
+            MethodRef stOf = MethodRef.method(JavaType.J_L_STRING_TEMPLATE, "of", JavaType.J_L_STRING_TEMPLATE,
+                    JavaType.J_U_LIST, JavaType.J_U_LIST);
+            Op.Result st = builder.op(invoke(stOf, fragmentsList, expressionsList));
+
+            MethodRef stpProcess = MethodRef.method(JavaType.J_L_STRING_TEMPLATE_PROCESSOR, "process",
+                    JavaType.J_L_OBJECT, JavaType.J_L_STRING_TEMPLATE);
+            Op.Result res = builder.op(invoke(resultType(), stpProcess, builder.context().getValue(processor()), st));
+
+            builder.context().mapValue(result(), res);
+
+            return builder;
+        }
+
+        private static Value buildArrayObject(Block.Builder builder, List<Value> elements) {
+            Value array = builder.op(newArray(J_L_OBJECT_ARRAY, builder.op(constant(JavaType.INT, elements.size()))));
+            for (int i = 0; i < elements.size(); i++) {
+                Value ele = elements.get(i);
+                if (isPrimitive(ele.type())) {
+                    TypeElement wt = getWrapperType(ele.type());
+                    MethodRef valueOf = MethodRef.method(wt, "valueOf", wt, ele.type());
+                    ele = builder.op(invoke(valueOf, ele));
+                }
+                builder.op(arrayStoreOp(array, builder.op(constant(JavaType.INT, i)), ele));
+            }
+            return array;
+        }
+
+        private static boolean isPrimitive(TypeElement te) {
+            return primitiveToWrapper().containsKey(te);
+        }
+
+        private static TypeElement getWrapperType(TypeElement te) {
+            return primitiveToWrapper().get(te);
+        };
+
+        private static Map<TypeElement, TypeElement> primitiveToWrapper() {
```

Comment:
Recommend moving this functionality to `JavaType`. (You can store the result of Map.of in a static final field and reuse that rather than recompute on each call.) 

When we improve the modeling of Java types we can improve further.